### PR TITLE
add password validation

### PR DESCRIPTION
**Pull request by britt6612**
## Work Completed 

[Preview](https://deploy-preview-983--keen-mayer-a86c8b.netlify.app/templates/forms)
[figma](https://www.figma.com/file/3fkwBelW5UsCbfwdDCJkT8/HPE-Form-Templates?node-id=0%3A1)

- Added password validation to match designs and new validation flow 
- Added this to both SignUp examples in form page

## To Do 

Greg is going to create a icon which will be used instead of `formclose` so once this icon is ready I will switch them out

### CODE REVIEW

File: aries-site/src/examples/templates/forms/SignUpExample.js [hunk 1]
```diff
@@ -147,9 +156,36 @@ export const SignUpExample = () => {
...
+                      return (
+                        <Box direction="row" gap="xsmall">
+                          <Text>{rule.message}</Text>
+                        </Box>
+                      );
+                    }
+                    return (
+                      <Box direction="row" gap="xsmall">
+                        {formValues.password && rule.valid ? (
+                          <FormCheckmark />
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SignUpExample.js:179]:
```suggestion
                          <FormCheckmark color="text-strong" />
```

File: aries-site/src/examples/templates/forms/SignUpExample.js [hunk 2]
```diff
@@ -147,9 +156,36 @@ export const SignUpExample = () => {
...
+                        ) : (
+                          <FormClose />
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SignUpExample.js:181]:
```suggestion
                          <FormClose color="text-strong" />
```

File: aries-site/src/examples/templates/forms/SignUpExample.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/examples/templates/forms/SignUpExample.js:179]: designs call for `text-strong` for icon

File: aries-site/src/examples/templates/forms/SignUpExample.js [hunk 3]
```diff
@@ -147,9 +156,36 @@ export const SignUpExample = () => {
...
+                        )}
+                        <Text>{rule.message}</Text>
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SignUpExample.js:183]:
```suggestion
                        <Text size="xsmall">{rule.message}</Text>
```

Comment (taysea) [reply aries-site/src/examples/templates/forms/SignUpExample.js:183]: Design calls for text `xsmall` (should be 14px)

File: aries-site/src/examples/templates/forms/SimpleSignUpExample.js [hunk 4]
```diff
@@ -147,7 +156,36 @@ export const SimpleSignUpExample = () => {
...
+                      return (
+                        <Box direction="row" gap="xsmall">
+                          <Text>{rule.message}</Text>
+                        </Box>
+                      );
+                    }
+                    return (
+                      <Box direction="row" gap="xsmall">
+                        {formValues.password && rule.valid ? (
+                          <FormCheckmark />
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SimpleSignUpExample.js:179]:
```suggestion
                          <FormCheckmark color="text-strong" />
```

File: aries-site/src/examples/templates/forms/SimpleSignUpExample.js [hunk 5]
```diff
@@ -147,7 +156,36 @@ export const SimpleSignUpExample = () => {
...
+                        ) : (
+                          <FormClose />
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SimpleSignUpExample.js:181]:
```suggestion
                          <FormClose color="text-strong" />
```

File: aries-site/src/examples/templates/forms/SimpleSignUpExample.js [hunk 6]
```diff
@@ -147,7 +156,36 @@ export const SimpleSignUpExample = () => {
...
+                        )}
+                        <Text>{rule.message}</Text>
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SimpleSignUpExample.js:183]:
```suggestion
                        <Text size="xsmall">{rule.message}</Text>
```

File: aries-site/src/examples/templates/forms/SignUpExample.js (hunk 2 above)
Comment (britt6612) [reply aries-site/src/examples/templates/forms/SignUpExample.js:181]: good catch

File: aries-site/src/examples/templates/forms/SimpleSignUpExample.js [hunk 7]
```diff
@@ -147,7 +156,36 @@ export const SimpleSignUpExample = () => {
...
+                  pad="none"
+                >
+                  {rule => {
+                    if (
+                      formValues.password === undefined ||
+                      formValues.password.length === 0
+                    ) {
+                      return (
+                        <Box direction="row" gap="xsmall">
+                          <Text>{rule.message}</Text>
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SimpleSignUpExample.js:172]:
```suggestion
                          <Text size="xsmall">{rule.message}</Text>
```

File: aries-site/src/examples/templates/forms/SignUpExample.js [hunk 8]
```diff
@@ -147,9 +156,36 @@ export const SignUpExample = () => {
...
+                  pad="none"
+                >
+                  {rule => {
+                    if (
+                      formValues.password === undefined ||
+                      formValues.password.length === 0
+                    ) {
+                      return (
+                        <Box direction="row" gap="xsmall">
+                          <Text>{rule.message}</Text>
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SignUpExample.js:172]:
```suggestion
                          <Text size="xsmall">{rule.message}</Text>
```

File: aries-site/src/examples/templates/forms/SignUpExample.js [hunk 9]
```diff
@@ -147,9 +156,36 @@ export const SignUpExample = () => {
...
+                      return (
+                        <Box direction="row" gap="xsmall">
+                          <Text size="xsmall">{rule.message}</Text>
+                        </Box>
+                      );
+                    }
+                    return (
+                      <Box direction="row" gap="xsmall">
+                        {formValues.password && rule.valid ? (
+                          <FormCheckmark color="text-strong" />
```
Comment (taysea) [inline aries-site/src/examples/templates/forms/SignUpExample.js:179]: Curious to see what these would look like as a smaller size -- the designs make it look like the icons are smaller

Comment (britt6612) [reply aries-site/src/examples/templates/forms/SignUpExample.js]: smaller size it is! This was approved on Greg as well

**Pull request merged**